tools/clang/unittests/HLSLExec/DXRUtil.h

```diff
@@ -0,0 +1,223 @@
+//===------------ DXRUtil.h - DXR Utility Functions ------------*- C++ -*-===//
+///////////////////////////////////////////////////////////////////////////////
+//                                                                           //
+// DXRUtil.h                                                                 //
+// Copyright (C) Nvidia Corporation. All rights reserved.                    //
+// This file is distributed under the University of Illinois Open Source     //
+// License. See LICENSE.TXT for details.                                     //
+//                                                                           //
+//  This file contains the utility functions for DXR execution tests.        //
+//                                                                           //
+///////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+//= DXR Utility
+//============================================================================
+#define SHADER_ID_SIZE_IN_BYTES 32
+
+#ifndef ROUND_UP
+#define ROUND_UP(v, PowerOf2Alignment)                                         \
+  (((v) + (PowerOf2Alignment)-1) & ~((PowerOf2Alignment)-1))
+#endif
+struct SceneConsts {
+  DirectX::XMFLOAT4 Eye;
+  DirectX::XMFLOAT4 U;
+  DirectX::XMFLOAT4 V;
+  DirectX::XMFLOAT4 W;
+  float SceneScale;
+  unsigned WindowSize[2];
+  int RayFlags;
+};
+
+struct Instance {
+  D3D12_RAYTRACING_GEOMETRY_TYPE Type;
+  DirectX::XMFLOAT4X4 Matrix;
+  UINT GeometryCount;
+  UINT BottomASIdx;
+  UINT InstanceID;
+  UINT Mask;
+  UINT Flags;
+};
+
+class ShaderTable {
+public:
+  void Init(ID3D12Device *Device, int RaygenCount, int MissCount,
+            int HitGroupCount, int RayTypeCount, int RootTableDwords) {
+    RayTypeCount = RayTypeCount;
```

Comment:
These initializers are no-ops because the left hand side and right hand side refer to the same argument variable. If this was an initializer list then it would work, or if the left hand side was accessed with `this->`.

Reply:
Made that a proper ctor with initializer list